t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -1240,7 +1240,16 @@ test_expect_success 'sparse index is not expanded: read-tree' '
 	do
 		ensure_not_expanded read-tree -mu $MERGE_TREES &&
 		ensure_not_expanded reset --hard HEAD
-	done
+	done &&
+
+	ensure_not_expanded checkout -b test-branch-2 &&
+	mkdir -p sparse-index/deep/test-dir &&
+	cp sparse-index/a sparse-index/deep/test-dir &&
+	ensure_not_expanded add deep/test-dir &&
+	ensure_not_expanded commit -m "test" &&
+
+	ensure_not_expanded checkout test-branch &&
+	ensure_not_expanded read-tree --prefix=deep/test-dir test-branch-2
```

Comment:
Do you want to try something with `--prefix=deep/deeper2` instead of creating this `test-dir`?

Reply:
When `read-tree`-ing with only one tree argument, `bind_merge(...)` prevents "merging" when a file already exists within the prefix. If I run `git read-tree --prefix=deep/deeper2 update-deep`, I get the error:

```
error: Entry 'deep/deeper2/a' overlaps with 'deep/deeper2/a'.  Cannot bind.
```

To avoid creating the new directory, though, I adjusted the test to instead remove the files in `deep/deeper2`, commit that to the test branch, then run `git read-tree --prefix=deep/deeper2 deepest`, which ends up testing the same(ish) scenario.